modules/cudastereo/src/cuda/stereobp.cu

```diff
@@ -260,7 +260,7 @@ namespace cv { namespace cuda { namespace device
             const int x = blockIdx.x * blockDim.x + threadIdx.x;
             const int y = blockIdx.y * blockDim.y + threadIdx.y;
 
-            if (x < dst_cols && y < dst_rows)
+            if (x < dst_cols && y < dst_rows && 2 * y + 1 < src_rows) // check the range strictly
```

Comment:
Perhaps, there is the similar problem is with "x" too.
Also dst element is stay uninitialized.

Try this approach:
```
if (...)
{
    const int x0 = 2*x;
    const int x1 = min(x0 + 1, src_cols - 1);
    const int y0 = 2*y;
    const int y1 = min(y0 + 1, src_rows - 1);
    for (int d ...)
    {
        float dst_reg  = src.ptr(d * src_rows + y0)[x0];
        dst_reg += src.ptr(d * src_rows + y1)[x0];
        dst_reg += src.ptr(d * src_rows + y0)[x1];
        dst_reg += src.ptr(d * src_rows + y1)[x1];
        ...
    }
}
```

Reply:
Thank you !
Yes, I've updated as you suggested.